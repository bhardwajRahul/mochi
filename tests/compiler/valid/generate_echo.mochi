let poem = generate text {
  prompt: "echo $word",
  args: { "word": "hello" }
}
print(poem)
