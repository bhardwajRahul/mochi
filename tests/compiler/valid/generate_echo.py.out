# Generated by Mochi Python compiler

def main():
	poem = _gen_text("echo $word", {"word": "hello"})
	print(poem)

def _gen_text(prompt, args):
    for k, v in args.items():
        placeholder = '$' + k
        prompt = prompt.replace(placeholder, str(v))
    return prompt
if __name__ == "__main__":
	main()
